### DISCUSSION TITLE
初始化项目，或者利用已有项目重新运行后，报错 Error: useSidebarItems() is called without provider.，有可能是哪里的问题？

### DISCUSSION THREAD

**Discussion by chenbihao** [vuepress/ecosystem#197] [Q&A] [answered]
新建项目 `npm init vuepress vuepress-starter` ，
或者利用旧项目（2.0.0-rc.2）删除package-lock文件后，重新运行，
或者使用 `vp-update`命令升级到最新版本，
都会得到以下结果：

![image](https://github.com/vuepress/core/assets/21167262/b876f719-fb9e-4564-8f3c-7968a0e2f13d)

这是我旧版的 `package.json` ，当没有删除原有 package-lock.json 时，它是能正常运行的，删除lock后重新install就不行了：
```json
{
  "name": "xdocs",
  "version": "0.0.1",
  "description": "A VuePress project",
  "license": "MIT",
  "type": "module",
  "homepage": "xxx",
  "scripts": {
    "docs:build": "vuepress build src",
    "docs:clean-dev": "vuepress dev src --clean-cache",
    "docs:dev": "vuepress dev src",
    "docs:update-package": "npx vp-update"
  },
  "devDependencies": {
    "@vuepress/client": "^2.0.0-rc.2",
    "@vuepress/bundler-vite": "^2.0.0-rc.2",
    "@vuepress/theme-default": "^2.0.0-rc.2",
    "vue": "^3.4.0",
    "vuepress": "^2.0.0-rc.2"
  }
}
```

由于我不是很熟悉前端内容，搞不清楚到底是哪里出现了问题，请求帮助

更多信息：
```
>node -v
v20.14.0
>npm -v
10.7.0
```
这是旧版可运行的
[old-package-lock.json](https://github.com/user-attachments/files/15546115/old-package-lock.json)
这是重新拉取后报错的
[new-package-lock.json](https://github.com/user-attachments/files/15546118/new-package-lock.json)

**Comment by Mister-Hope** [comment]
give me a full reproduction on stackblitz

**Reply by chenbihao** [reply]
https://stackblitz.com/edit/stackblitz-starters-6vfpe1?file=package.json

**Reply by Mister-Hope** [reply]
Thanks for the reproduction. I will review that later

**Reply by Mister-Hope** [reply, marked as answer]
A bug, fixed in https://github.com/vuepress/ecosystem/commit/e5dcba2c1ad03ef3083fcfc3bcc8d0105793f270, thanks for reporting.
